File: config/routes/api/v2/stats.rb
# frozen_string_literal: true

resource :stats, only: [:show] do
  get :models_per_month, path: "models-per-month"
  get :models_by_size, path: "models-by-size"
  get :models_by_production_status, path: "models-by-production-status"
  get :models_by_manufacturer, path: "models-by-manufacturer"
  get :models_by_classification, path: "models-by-classification"
  get :components_by_class, path: "components-by-class"
end

File: app/frontend/frontend/api/collections/Stats.ts
import { get } from "@/frontend/api/client";
import type { TApiResponse, TApiErrorResponse } from "@/frontend/api/client";

export class StatsCollection {
  quickStats(): Promise<TApiResponse<TQuickStats> | TApiErrorResponse> {
    return get<TQuickStats>("stats/quick-stats");
  }

  modelsByClassification(): Promise<
    TApiResponse<TChartStats[]> | TApiErrorResponse
  > {
    return get<TChartStats[]>("stats/models-by-classification");
  }

  modelsBySize(): Promise<TApiResponse<TChartStats[]> | TApiErrorResponse> {
    return get<TChartStats[]>("stats/models-by-size");
  }

  modelsPerMonth(): Promise<
    TApiResponse<TBarChartStats[]> | TApiErrorResponse
  > {
    return get<TBarChartStats[]>("stats/models-per-month");
  }

  modelsByManufacturer(): Promise<
    TApiResponse<TChartStats[]> | TApiErrorResponse
  > {
    return get<TChartStats[]>("stats/models-by-manufacturer");
  }

  modelsByProductionStatus(): Promise<
    TApiResponse<TChartStats[]> | TApiErrorResponse
  > {
    return get<TChartStats[]>("stats/models-by-production-status");
  }
}

export default new StatsCollection();
